src/server/modes/charts/plugins/datalens/preparers/bar-x/prepareBarX.ts

```diff
@@ -453,13 +456,13 @@ export function prepareBarX(args: PrepareFunctionArgs) {
             });
         });
 
-        if (isColorizeByMeasure || isColorizeByMeasureValue) {
-            colorizeByMeasure(visualizationId as WizardVisualizationId, {
+        if (colorMode === ColorMode.GRADIENT || isColorizeByMeasure || isColorizeByMeasureValue) {
```

Comment:
It is better to put it in a separate utility (duplicated in several more places by code)